components/omega/test/ocn/EosTest.cpp

```diff
@@ -285,6 +291,117 @@ int testEosTeos10Displaced() {
    return Err;
 }
 
+int testEosTeos10Clamping() {
+   int Err          = 0;
+   const auto *Mesh = HorzMesh::getDefault();
+   /// Get Eos instance to test
+   Eos *TestEos       = Eos::getInstance();
+   TestEos->EosChoice = EosType::Teos10Eos;
+
+   /// Create and fill ocean state arrays
+   Array2DReal SArray = Array2DReal("SArray", Mesh->NCellsAll, NVertLevels);
+   Array2DReal TArray = Array2DReal("TArray", Mesh->NCellsAll, NVertLevels);
+   Array2DReal PArray = Array2DReal("PArray", Mesh->NCellsAll, NVertLevels);
+   /// Use Kokkos::deep_copy to fill the entire view with the ref value
+   /// Test with valid poly75t values first.
+   deepCopy(SArray, 35.0);
+   deepCopy(TArray, 5.0);
+   deepCopy(PArray, 400.0);
+   deepCopy(TestEos->SpecVol, 0.0);
+
+   /// Compute specific volume
+   TestEos->computeSpecVol(TArray, SArray, PArray);
+
+   /// Check all array values against expected value
+   int numMismatches   = 0;
+   Array2DReal SpecVol = TestEos->SpecVol;
+   parallelReduce(
+       "CheckSpecVolMatrix-Teos", {Mesh->NCellsAll, NVertLevels},
+       KOKKOS_LAMBDA(int i, int j, int &localCount) {
+          if (!isApprox(SpecVol(i, j), TeosClampValue1, RTol)) {
+             localCount++;
+          }
+       },
+       numMismatches);
+
+   auto SpecVolH = createHostMirrorCopy(SpecVol);
+   if (numMismatches != 0) {
+      Err++;
+      LOG_ERROR("EosTest: TEOS SpecVolClampingNone isApprox FAIL, "
+                "expected {}, got {} with {} mismatches",
+                TeosClampValue1, SpecVolH(1, 1), numMismatches);
+   }
+   if (Err == 0) {
+      LOG_INFO("EosTest SpecVolClampingNone TEOS-10: PASS");
+   }
+
+   /// Test with an ivalid poly75t salinity value second
+   deepCopy(SArray, 45.0);
+   deepCopy(TArray, 5.0);
+   deepCopy(PArray, 400.0);
+   deepCopy(TestEos->SpecVol, 0.0);
+
+   /// Compute specific volume
+   TestEos->computeSpecVol(TArray, SArray, PArray);
+
+   /// Check all array values against expected value
+   numMismatches = 0;
+   SpecVol       = TestEos->SpecVol;
+   parallelReduce(
+       "CheckSpecVolMatrix-Teos", {Mesh->NCellsAll, NVertLevels},
+       KOKKOS_LAMBDA(int i, int j, int &localCount) {
+          if (!isApprox(SpecVol(i, j), TeosClampValue2, RTol)) {
+             localCount++;
+          }
+       },
+       numMismatches);
+
+   SpecVolH = createHostMirrorCopy(SpecVol);
+   if (numMismatches != 0) {
+      Err++;
+      LOG_ERROR("EosTest: TEOS SpecVolClampingSalt isApprox FAIL, "
+                "expected {}, got {} with {} mismatches",
+                TeosClampValue2, SpecVolH(1, 1), numMismatches);
+   }
+   if (Err == 0) {
+      LOG_INFO("EosTest SpecVolClampingSalt TEOS-10: PASS");
+   }
+
+   /// Test with an ivalid poly75t temperature value third
+   deepCopy(SArray, 35.0);
+   deepCopy(TArray, 100.0);
+   deepCopy(PArray, 400.0);
+   deepCopy(TestEos->SpecVol, 0.0);
```

Comment:
Should we add a test for an invalid low temperature?

Reply:
Sure, or we could replace the value for TArray to be low instead of high, since it exercises the freezing point code (which has its own test, but still). 
We are not testing all out-of-bound combinations, this was mostly a check on whether clamping was applied properly but maybe it's most bang for our buck to apply to the low T than the high T.
Or would you prefer to see both tests for T @sbrus89?

Reply:
Done!